atomix/cluster/src/main/java/io/atomix/raft/roles/AbstractAppender.java

```diff
@@ -447,45 +449,63 @@ protected void handleConfigureResponseError(
     // This prevents infinite loops when cluster configurations fail.
   }
 
-  /** Builds an install request for the given member. */
-  protected InstallRequest buildInstallRequest(
+  /**
+   * Builds an install request for the given member.
+   *
+   * @return
+   */
+  protected Optional<InstallRequest> buildInstallRequest(
       final RaftMemberContext member, final Snapshot snapshot) {
     if (member.getNextSnapshotIndex() != snapshot.index()) {
+      try {
+        final SnapshotChunkReader snapshotChunkReader = snapshot.newChunkReader();
+        member.setSnapshotChunkReader(snapshotChunkReader);
+      } catch (final UncheckedIOException e) {
+        log.warn(
+            "Expected to send Snapshot {} to {}. But could not open SnapshotChunkReader. Will retry.",
+            snapshot.getPath(),
+            e);
+        return Optional.empty();
+      }
       member.setNextSnapshotIndex(snapshot.index());
       member.setNextSnapshotChunk(null);
     }
 
     final InstallRequest request;
-    synchronized (snapshot) {
```

Comment:
I removed this `synchronized`. I think it is not required. What do you think? @Zelldon

Reply:
I think the same, since we having only one thread.

Reply:
:+1: